BGL/include/CGAL/boost/graph/named_params_helper.h

```diff
@@ -112,8 +113,156 @@ namespace CGAL {
     typedef typename boost::property_traits<PMap>::value_type type;
   };
 
+  namespace internal_np
+  {
+  template<typename key_value, class PMap, class Graph, typename Tag>
+  struct MapInitializer{
+    void operator()(PMap, const Graph& )
+    {}
+  };
+
+  template<typename key_value, class PMap, class Graph>
+  struct MapInitializer<key_value, PMap, Graph, CGAL::Tag_false>{
+    void operator()( PMap, const Graph& )
+    {
+      //do nothing.
+    }
+  };
+
+  template< class PMap, class Graph>
+  struct MapInitializer<
+      typename boost::graph_traits<Graph>::vertex_descriptor,
+      PMap, Graph,
+      CGAL::Tag_true>{
+    void operator()(PMap map, const Graph& g)
+    {
+      CGAL::helpers::init_vertex_indices(g, map);
+    }
+  };
+
+  template< class PMap, class Graph>
+  struct MapInitializer<
+      typename boost::graph_traits<Graph>::halfedge_descriptor,
+      PMap, Graph,
+      CGAL::Tag_true>{
+    void operator()(PMap map, const Graph& g)
+    {
+      CGAL::helpers::init_halfedge_indices(g, map);
+    }
+  };
+
+  template< class PMap, class Graph>
+  struct MapInitializer<
+      typename boost::graph_traits<Graph>::face_descriptor,
+      PMap, Graph,
+      CGAL::Tag_true>{
+    void operator()(PMap map,const Graph& g)
+    {
+      CGAL::helpers::init_face_indices(g, map);
+    }
+  };
+
+  template<typename PMapCategory>
+  struct Is_pmap_writable{
+    typedef CGAL::Tag_false result;
+  };
+
+  template<>
+  struct Is_pmap_writable<boost::read_write_property_map_tag>{
+    typedef CGAL::Tag_true result;
+  };
+
+  template<>
+  struct Is_pmap_writable<boost::writable_property_map_tag>{
```

Comment:
you're missing mutable lvalue pmap, see [here](https://github.com/CGAL/cgal/blob/master/BGL/include/CGAL/boost/graph/properties.h#L126) (you can also probably save space with a macro if you want). BTW writable tag does not really makes sense if you can't read them...

Reply:
Yes but a function `is_writable()` that reeturns false for `writable_tag` doesn't seem very trustable :)